### DISCUSSION TITLE
Implement Sqids along Hashids

### DISCUSSION THREAD

**Discussion by degj21** [ash-jc-allen/short-url#299] [Ideas]
Hi there! Searching for a short URL generator I found this package and it works like a charm. However, I noticed the original dependency of hashids/hashids has been rebranded and upgraded to this one: [https://github.com/sqids/sqids-php](url). Could it be possible to incorporate the rebranded package as a dependency, maintaining the compatibility with Hashids? Thanks in advance for your feedback!

**Comment by ash-jc-allen** [comment]
Hey! Thanks, I'm glad you're liking the package 😄

Ahh I didn't know it had been rebranded. Do you know if there's an announcement or anything?

In the theory, I think we should be able to add a new driver that implements the `AshAllenDesign\ShortURL\Interfaces\UrlKeyGenerator`. Then anyone using the project could use that driver by setting the `short-url.url_key_generator` config field.

I'm happy to review any PRs to contribute this 😄

**Comment by degj21** [comment]
Thanks for your answer Ash! 😄 In fact, they announced the rebranding in their Github profile [https://github.com/hashids](url).

It is interesting that they changed the algorithm and mostly rebuild and upgrade the original library, that is still available, but all its previous links now redirect here: [https://sqids.org/](url). Can you please take a look at the new implementation and keep me updated about whether it is possible to implement it? 🙏​ Also, I don't think it is a bad idea to consider dropping Hashids if it is required. Whatever suits better, I'll be glad to learn in this process (mostly driver creation and Interface implementation) and collaborate wherever possible! ​🤝  (By the way, I'm not a Git expert, still learning about pull requests and all that stuff 😅). Best regards! 👋​

**Reply by ash-jc-allen** [reply]
Thanks for the link! I'm glad you pointed out the announcement because I hadn't heard any mention of it haha.

From having a quick look, I think we might be able to add a new driver to support the new library. But I don't think it'll provide all the functionality that we've currently got when using Hashids. So I think I'll probably keep Hashids in there as well for as long as possible (until it starts causing issues when installing Composer dependencies haha!) 😄

I'm pretty snowed under at the moment, so it's probably something I'll take a look at over Christmas as a fun little task. But if anyone else wants to have a crack at it before then, that's cool with me!

**Reply by degj21** [reply]
Hi there! Hoping that you all have started the year strong. Are there any updates concerning the Hashids/Sqids implementation? I'll be on the watch for this. Best regards!

Daniel

**Reply by ash-jc-allen** [reply]
Hey @degj21! Likewise, I hope the new year's going well so far 😄 Unfortunately, I've not had a chance to get stuck into this yet, sorry :(

**Reply by degj21** [reply]
Hello Ash, hope you're having a nice 2025 so far. Although there haven't been issues with hashids (repo is still available), I'm wondering if that will be always the case. Sqids has been  around for quite a while and looks good too. I'm wondering if, instead of replacing Hashids, Squids could be put along Hashids, like a new way to generate the short URLs. So if Hashids ever gets removed, Sqids could be the default. Best regards!
